linearregression/Lib.idr

```diff
@@ -0,0 +1,51 @@
+module Lib
+import Types
+
+
+public export
+calculateDeltas : (Double,Double) -> (Double, Double) -> Double
+calculateDeltas (t0, t1) (x, y) = t0 + t1 * x-y
+
+
+
+avg : List Double -> Double
+avg [] = 0.0
+avg xs = do
+  let x = length xs
+  let y = natToInteger x
+  let z = fromInteger y
+  (sum xs) / z
+
+
+
+getmult: List Double -> List Double -> List Double
+getmult [] [] = []
+getmult (x::xs) [] = (x::getmult xs [])
+getmult [] (y::ys) = (y::getmult [] ys)
+getmult (x::xs) (y::ys) = (x * y :: getmult xs ys)
```

Comment:
I suspect this can be simplified using some higher lever function like `map`, but I don't know which exactly, maybe @sroberti knows.

Reply:
This is indeed the case.
`getmult = zipWith (*)` does the trick for me (though if you're using Idris 2, you'll have to import Data.List for it to work).

Reply:
True, Haskell has better doc than Idris.  Also for Haskell we have the excellent https://hoogle.haskell.org/. For idris we have `:s` in the REPL but it doesn't always work as expected, for instance I tried the following

```
:s (a -> b -> c) -> List a -> List b -> List c
```

using the type I got with

```
:t zipWith
```

hoping to get `zipWith` in return but got nothing (and I did load the Data.List module before that).

Anybody, any idea why it doesn't work?

Reply:
Yes I  noticed that too @ngeiswei! However, I believe the :s works fine in idris1.  while its faster and have new modules idris2 kind of sucks on this kind of things that makes development easier. idris 1 on the other hand  is so development friendly. Did anybody else encountered/noticed tab completion  problems in The REPL for idris2?

Reply:
Anyways I have both versions of idris installed so when I do search on REPL I use the older version if that helps @ngeiswei

Reply:
Agreed, idris2 is not as dev friendly (lack of code completion, etc), so we can use idris1 for searching or anything that can compensate idris2 for.  However I think we should pursue using idris2 for developing the ai-dsl, simply because it's type system is more capable, including things that we need, see that PR https://github.com/singnet/ai-dsl/pull/2 (unless I'm mistaken and idris1 can do that).